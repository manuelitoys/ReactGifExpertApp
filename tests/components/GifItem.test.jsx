import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => { 
    
    const title = 'Saitama';
    const url = 'https://saitama.com/'

    test('should make match with Snapshot', () => { 


        const { container } = render( <GifItem title={ title } url={ url } /> )
        expect( container ).toMatchSnapshot()

     });

     test('should show the image with url right', () => { 

        render( <GifItem title={ title } url={ url } /> )

        // expect( screen.getByRole('img').src ).toBe( url )
        // expect( screen.getByRole('img').alt ).toBe( title )

        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url )
        expect( alt ).toBe( title )

      })

      test('should show title of component', () => { 

        render( <GifItem title={ title } url={ url } /> )
        expect( screen.getByText( title ) ).toBeTruthy()

       })

 })