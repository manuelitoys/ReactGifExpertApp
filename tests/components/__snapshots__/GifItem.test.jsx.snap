// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should make match with Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      role="img"
      src="https://saitama.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
